sdk/storage/Azure.Storage.Blobs/src/BlobContainerClient.cs

```diff
@@ -1663,17 +1663,26 @@ private void SetNameFieldsIfNull()
 
                 try
                 {
-                    return await BlobRestClient.Container.ListBlobsFlatSegmentAsync(
-                        ClientDiagnostics,
-                        Pipeline,
-                        Uri,
-                        marker: marker,
-                        prefix: prefix,
-                        maxresults: pageSizeHint,
-                        include: BlobExtensions.AsIncludeItems(traits, states),
-                        async: async,
-                        cancellationToken: cancellationToken)
-                        .ConfigureAwait(false);
+                    Response<BlobsFlatSegment> response = await BlobRestClient.Container.ListBlobsFlatSegmentAsync(
+                          ClientDiagnostics,
+                          Pipeline,
+                          Uri,
+                          marker: marker,
+                          prefix: prefix,
+                          maxresults: pageSizeHint,
+                          include: BlobExtensions.AsIncludeItems(traits, states),
+                          async: async,
+                          cancellationToken: cancellationToken)
+                          .ConfigureAwait(false);
+                    if (!traits.Equals(BlobTraits.Metadata))
```

Comment:
Wouldn't we want to use (traits & BlobTraits.Metadata) != BlobTraits.Metadata
since this is a Flags enum? This is the equivalent of using the HasFlags method but we don't want to use that due to perf.